presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -496,6 +506,149 @@ public CounterStat getHighMemorySplitSource()
         return concat(partitionBatches);
     }
 
+    private static PartitionBatchInfo getPartitionBatchInfo(
+            ConnectorSession session,
+            SemiTransactionalHiveMetastore metastore,
+            SchemaTableName tableName,
+            List<HivePartition> partitionBatch,
+            Map<String, HiveColumnHandle> predicateColumns,
+            Optional<Map<Subfield, Domain>> domains)
+    {
+        if (isPartitionStatsBasedOptimizationEnabled(session) && domains.isPresent()) {
```

Comment:
I think we can remove the if condition here. if we have `isPartitionStatsBasedOptimizationEnabled(session)`, then we can create an empty map for partitionStatistics. Otherwise call `getPartitionStatistics`

Reply:
I changed the logic a bit, now it looks like:
```
partitions = <get partitions by names>;
partitionStatistics = ImmutableMap.of();
if (domain.isPresent() && <session property is true>) {
  partitionStatistics = <get partition statistics>
}
<process partitions>
```